### PR TITLE
Smartinez us2

### PR DESCRIPTION
**Pull request by smartinezmurial**
Waiting for code review

### CODE REVIEW

File: compensations/debug/logs/2017-03-24 09-42-44.log [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+09:42:42.0 (5710103)|VARIABLE_ASSIGNMENT|[5]|this.recordType|null|0x1acc0b80
+09:42:42.0 (5756541)|SYSTEM_MODE_ENTER|true
+09:42:42.0 (5782686)|HEAP_ALLOCATE|[EXTERNAL]|Bytes:16
+09:42:42.0 (5787119)|VARIABLE_SCOPE_BEGIN|[17]|this|CompensationController|true|false
+09:42:42.0 (5804246)|VARIABLE_ASSIGNMENT|[17]|this|{"employeeId":"00346000002uvVF","filter":""}|0x1acc0b80
+09:42:42.0 (5811816)|STATEMENT_EXECUTE|[17]
+09:42:42.0 (5813171)|STATEMENT_EXECUTE|[18]
+09:42:42.0 (5815462)|HEAP_ALLOCATE|[19]|Bytes:183
+09:42:42.0 (5819669)|HEAP_ALLOCATE|[21]|Bytes:2
+09:42:42.0 (5822271)|HEAP_ALLOCATE|[23]|Bytes:12
```
Comment (rjulianialtimetrik) [inline compensations/debug/logs/2017-03-24 09-42-44.log]: Please... DO NOT ADD Logs to version control!

File: compensations/src/classes/CompensationController.cls [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        );
+        return comps;
+    }
+    
+    public void cleanFilters(){
+        filter = '';
+    }
+    
+    public void filterStudyCompensations(){
+        filter = 'WHERE RecordType.Name = \'Study\' ';
```
Comment (rjulianialtimetrik) [inline compensations/src/classes/CompensationController.cls]: I don't think this actually works. if any of the filters are applied, the query would have 2 "WHERE" clauses. Please review this construction ASAP.

File: compensations/src/classes/TestCompensationController.cls [hunk 3]
```diff
@@ -0,0 +1,251 @@
+@isTest
+private class TestCompensationController {
+    
+    @isTest static void testGetCompensations(){
+        Contact c = new Contact();
```
Comment (rjulianialtimetrik) [inline compensations/src/classes/TestCompensationController.cls]: Ideally, data setup should be done in test setup methods. Please see the [apex guide](https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_testing_testsetup_using.htm) for more info on that.

File: compensations/src/classes/TestLockCompensationStatus.cls [hunk 4]
```diff
@@ -0,0 +1,100 @@
+@isTest
+private class TestLockCompensationStatus {
+    
+    @isTest static void testTriggerOnDelete(){
+        Contact c = new Contact();
+        c.FirstName = 'Juan';
```
Comment (rjulianialtimetrik) [inline compensations/src/classes/TestLockCompensationStatus.cls]: Again, as mentioned before, please use test setup methods for data generation.

File: compensations/src/classes/CompensationController.cls (hunk 2 above)
Comment (smartinezmurial) [inline compensations/src/classes/CompensationController.cls]: For each filter I have 2 functions, one of them is used when the compensations are not filtered by user, so the filter in that case is "WHERE", and there is another function for when the compensation are filtered by employee and in that case the filter is "AND"

File: compensations/src/classes/TestCompensationController.cls (hunk 3 above)
Comment (smartinezmurial) [inline compensations/src/classes/TestCompensationController.cls]: Done

File: compensations/src/classes/TestLockCompensationStatus.cls (hunk 4 above)
Comment (smartinezmurial) [inline compensations/src/classes/TestLockCompensationStatus.cls]: Done

**Pull request merged**